frontend/src/features/frame/components/topbar/TopBar.tsx

```diff
@@ -140,14 +152,41 @@ const TopBar = () => {
           padding: '4px !important',
         }}
       >
-        <Grid container sx={{ width: '100%' }}>
-          <Logo />
+        <Grid container sx={{ width: '100%', maxWidth: '100%' }}>
           <Hidden mdDown>
+            <Logo />
             <Grid item md={4}>
               {options?.withSearchBar && <Search />}
             </Grid>
+            <AccountInfo />
+          </Hidden>
+          <Hidden mdUp>
+            {!mobileSearchOpen && <Logo />}
+            {!mobileSearchOpen && (
+              <IconButton
+                aria-label="Open the searchbar"
+                onClick={() => searchOpeningHandler(true)}
+              >
+                <SearchIcon fontSize="large" />
+              </IconButton>
+            )}
+            {mobileSearchOpen && (
+              <Grid container width="100%">
+                <Grid item xs={11}>
+                  {options?.withSearchBar && <Search />}
+                </Grid>
+                <Grid item xs={1}>
+                  <IconButton
+                    aria-label="Close the searchbar"
+                    onClick={() => searchOpeningHandler(false)}
+                  >
+                    <CloseIcon />
+                  </IconButton>
+                </Grid>
+              </Grid>
+            )}
+            {!mobileSearchOpen && <AccountInfo />}
```

Comment:
Code Style:

I would find it easier to read with a ternary operator rather than repeating `!mobileSearchOpen && ...`. With the code below, I see more easily what's there when `mobileSearchOpen` is `true`or when it's `false`

```
            {mobileSearchOpen 
             ?
             <Grid container width="100%">
                <Grid item xs={11}>
                  {options?.withSearchBar && <Search />}
                </Grid>
                <Grid item xs={1}>
                  <IconButton
                    aria-label="Close the searchbar"
                    onClick={() => searchOpeningHandler(false)}
                  >
                    <CloseIcon />
                  </IconButton>
                </Grid>
              </Grid>
              : <>
                <Logo />
                <IconButton
                aria-label="Open the searchbar"
                onClick={() => searchOpeningHandler(true)}
              >
                <SearchIcon fontSize="large" />
              </IconButton>
              <AccountInfo />
               </>
              }
```